go-chaos/cmd/cluster.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2023 Camunda Services GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/zeebe-io/zeebe-chaos/go-chaos/internal"
+)
+
+func AddClusterCommands(rootCmd *cobra.Command, flags *Flags) {
+	var clusterCommand = &cobra.Command{
+		Use:   "cluster",
+		Short: "Interact with the Cluster API",
+		Long:  "Can be used query cluster topology and to request dynamic scaling",
+	}
+	var statusCommand = &cobra.Command{
+		Use:   "status",
+		Short: "Queries the current cluster topology",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return printCurrentTopology(flags)
+		},
+	}
+	var waitCommand = &cobra.Command{
+		Use:   "wait",
+		Short: "Waits for a topology change to complete",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return waitForChange(flags)
+		},
+	}
+
+	rootCmd.AddCommand(clusterCommand)
+	clusterCommand.AddCommand(statusCommand)
+	clusterCommand.AddCommand(waitCommand)
+	waitCommand.Flags().IntVar(&flags.changeId, "changeId", 0, "The id of the change to wait for")
```

Comment:
:upside_down_face: I know almost all Go linters complain about using `Id`, and always want to change to `ID`. While I have no strong feelings about it, if we ever add them I imagine this will happen :smile: 

I guess that applies to all the `Id` you have below. :shrug: 

```suggestion
	waitCommand.Flags().IntVar(&flags.changeID, "changeId", 0, "The id of the change to wait for")
```

Reply:
So far I haven't seen a linter complain about `bpmnProcessId`, `partitionId` etc. I really dislike `changeID` so I think I'll stick with `changeId` for now